src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -125,12 +123,12 @@ public ServiceEndpoint(string connectionString, EndpointType type = EndpointType
         var result = ConnectionStringParser.Parse(connectionString);
         AccessKey = result.AccessKey;
         _serviceEndpoint = result.Endpoint;
-        ClientEndpoint = result.ClientEndpoint;
-        ServerEndpoint = result.ServerEndpoint;
         EndpointType = type;
         Name = name;
 
         Endpoint = BuildEndpointString(_serviceEndpoint);
+        ClientEndpoint = result.ClientEndpoint ?? _serviceEndpoint;
+        ServerEndpoint = result.ServerEndpoint ?? _serviceEndpoint;
         AudienceBaseUrl = BuildAudienceBaseUrlEndWithSlash(_serviceEndpoint);
```

Comment:
either:
```
public Uri ClientEndpoint
    {
        get => _clientEndpoint ?? _serviceEndpoint; 
    }
```

    and here: `_clientEndpoint = CheckScheme(result.ClientEndpoint);`

Or:
```

public Uri ClientEndpoint{get; init;}
```

    and here: `ClientEndpoint = CheckScheme(result.ClientEndpoint) ?? _serviceEndpoint;`
?